### PR TITLE
Drop "userPassword" from Search Results

### PR DESCRIPTION
**Pull request by rhafer**
Let's not expose the password hashes to clients.

### CODE REVIEW

**Comment by rhafer** [comment]
ping @longsleep, would you mind giving this a review.

File: server/handler/boltdb/handler.go [hunk 1]
```diff
@@ -207,8 +207,15 @@ func (h *boltdbHandler) Search(boundDN string, req *ldap.SearchRequest, conn net
 
 	entries, _ := h.bdb.Search(req.BaseDN, req.Scope)
 
+	// strip "userPassword" Attribute from the Results
+	var sr []*ldap.Entry
+	for _, e := range entries {
+		entry := ldapentry.EntryDropAttribute(e, "userPassword")
```
Comment (longsleep) [inline server/handler/boltdb/handler.go:213]: Why is this only done for the bolt handler? Wouldn't it be better to deal with this on the database level? I don't remember how I dealt with this in the Ldif implementation.

Comment (longsleep) [reply server/handler/boltdb/handler.go:213]: https://github.com/libregraph/idm/blob/8d51161be79569f00dd239ca3f8c6ad9b840ee44/server/handler/ldif/entry.go#L19 ah here - it's stored separately so its not returned, no extra computation required. I would prefer so see something similar for the bolt backend so it does not need any computation in the hot request path.

**Comment by rhafer** [comment]
Closing for now.

**Pull request closed without merging**